src/status_im/ui2/screens/chat/messages/pinned_message.cljs

```diff
@@ -0,0 +1,49 @@
+(ns status-im.ui2.screens.chat.messages.pinned-message
+  (:require [status-im.i18n.i18n :as i18n]
+            [quo.react :as react]
+            [quo2.reanimated :as reanimated]
+            [quo.react-native :as rn]
+            [quo2.foundations.typography :as typography]
+            [quo2.foundations.colors :as quo2.colors]
+            [status-im.switcher.constants :as constants]
+            [status-im.chat.models.pin-message :as models.pin-message]
+            [status-im.utils.handlers :refer [<sub >evt]]
+            [status-im.ui2.screens.chat.messages.style :as style]
+            [quo2.components.icon :as icons]))
+
+(defn pin-limit-popover [chat-id pinned-messages-list]
+  [:f>
+   (fn []
+     (let [{:keys [width]} (constants/dimensions)
+           show-pin-limit-modal? (<sub [:chats/pin-modal chat-id])
+           opacity-animation (reanimated/use-shared-value 0)
+           z-index-animation (reanimated/use-shared-value -1)]
+       (react/effect! #(do
+                         (reanimated/set-shared-value opacity-animation (reanimated/with-timing (if show-pin-limit-modal? 1 0)))
+                         (reanimated/set-shared-value z-index-animation (reanimated/with-timing (if show-pin-limit-modal? 10 -1)))))
+       [reanimated/view {:style (reanimated/apply-animations-to-style
+                                 {:opacity opacity-animation
+                                  :z-index z-index-animation}
+                                 (style/pin-popover width))}
+        [rn/view {:style (style/pin-alert-container)}
```

Comment:
just wondering if it should be an icon

Reply:
Hmm, a text component is simpler I think